uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/feature/UserTokenGrantIT.java

```diff
@@ -0,0 +1,111 @@
+package org.cloudfoundry.identity.uaa.integration.feature;
+
+import org.cloudfoundry.identity.uaa.test.UaaTestAccounts;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.openqa.selenium.WebDriver;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.oauth2.client.test.TestAccounts;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.web.client.RestOperations;
+
+import java.util.Collections;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = DefaultIntegrationTestConfig.class)
+public class UserTokenGrantIT {
+
+  @Autowired
+  @Rule
+  public IntegrationTestRule integrationTestRule;
+
+  @Autowired
+  WebDriver webDriver;
+
+  @Value("${integration.test.base_url}")
+  String baseUrl;
+
+  @Value("${integration.test.app_url}")
+  String appUrl;
+
+  @Autowired
+  RestOperations restOperations;
+
+  @Autowired
+  TestClient testClient;
+
+  @Autowired
+  TestAccounts testAccounts;
+
+  final String user_token_id = "oauth_showcase_user_token";
+  final String user_token_secret = "secret";
+
+  @Before
+  @After
```

Comment:
Really like seeing the @Before and @After together here for cleanup.

Reply:
after is enough

Reply:
I think that's the first time I saw code being changed because I praised it. :)